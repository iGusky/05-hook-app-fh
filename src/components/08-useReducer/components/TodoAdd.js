import React from 'react'
import { useForm } from '../../../hooks/useForm';

export const TodoAdd = ({ handleAddTodo }) => {

    const [ {description}, handleInputChange, reset ] = useForm({
        description: ''
    });

    const handleSubmit = (e) => {
        e.preventDefault();

        if( description.trim().length > 1 ){          
            const newTodo = {
                id: new Date().getTime(),
                desc: description,
                done: false
            };

            handleAddTodo( newTodo );
            reset();
        }
    };

    return (
        <>
          <h4 className = "text-center">Agregar TODO</h4>
                    <form onSubmit = { handleSubmit }>
                       <div className = "form-group">
                        <input
                                type = "text"
                                name = "description"
                                placeholder = "Aprender..."
                                autoComplete = "off"
                                className = "form-control"
                                value = { description }
                                onChange = { handleInputChange }
                            />
                            <button
                                type = "submit"
                                className = "btn btn-outline-primary btn-block mt-2"
                            >
                                Agregar
                            </button>
                       </div>
                    </form>  
        </>
    )
}
