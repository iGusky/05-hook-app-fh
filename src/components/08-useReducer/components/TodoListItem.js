import React from 'react'

export const TodoListItem = ({todo, index:i, handleDelete, handleToggle}) => {
    return (
        <>
            <li 
                key = { todo.id }
                className = "list-group-item"
                >
                <p 
                    className = {`${ todo.done && 'complete'}`}
                    onClick = { () => handleToggle(todo.id) }    
                >
                    { i }. { todo.todo }
                </p>
                <button 
                    className = "btn btn-danger"
                    onClick = { () => handleDelete(todo.id) }    
                >
                    borrar
                </button>
            </li>
        </>
    )
}
