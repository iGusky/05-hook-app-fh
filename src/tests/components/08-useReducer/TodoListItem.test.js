import { TodoListItem } from "../../../components/08-useReducer/components/TodoListItem"
import { shallow } from 'enzyme';
import { demoTodos } from "../../fixtures/demoTodos";

describe('Pruebas en <TodoListItem />', () => {
    
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow(<TodoListItem
        todo = { demoTodos[0] }
        index = { demoTodos[0].id }
        handleDelete = { handleDelete }
        handleToggle = { handleToggle }
    />);

    test('debe mostrarse correctamente', () => {
        expect ( wrapper ).toMatchSnapshot();
    });

    test('debe de llamar la funciÃ³n handleDelete', () => {

        wrapper.find('button').simulate('click')
        expect( handleDelete ).toHaveBeenCalledWith( demoTodos[0].id );
    });

    test('debe de llamar la funcion handleToggle', () => {
        wrapper.find('p').simulate('click');
        expect( handleToggle ).toHaveBeenCalledWith( demoTodos[0].id );
    });

    test('debe mostrar el texto correctamente', () => {
        const test = wrapper.find('p').text();
        expect( test ).toBe(`${demoTodos[0].id}. ${demoTodos[0].todo}`);
    });

    test('debe tener la clase COMPLETE si el TODO.DONE es TRUE', () => {
        const todo = demoTodos[0];
        todo.done = true;
        const wrapper = shallow(<TodoListItem todo={todo}/>);
        expect( wrapper.find('p').hasClass('complete') ).toBe(true);
    })
    
    
    
    
    

})
