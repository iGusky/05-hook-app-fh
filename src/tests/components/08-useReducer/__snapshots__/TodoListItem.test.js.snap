// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoListItem /> debe mostrarse correctamente 1`] = `
<Fragment>
  <li
    className="list-group-item"
    key="1"
  >
    <p
      className="false"
      onClick={[Function]}
    >
      1
      . 
      Aprender Node
    </p>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      borrar
    </button>
  </li>
</Fragment>
`;
